library/firewall_lib.py

```diff
@@ -777,9 +827,14 @@ def main():
                 " port, source_port, protocol, destination, or helper_module)"
             )
 
-        # Zone and service are incompatible when state is set to present or absent
-        if zone and service:
-            module.fail_json(msg="both zone and service while state present/absent")
+        # Zone, service, and ipset are incompatible when state is set to present or absent
+        num_conflicting_args = len([x for x in [zone, ipset, service] if x])
```

Comment:
I think the code would be much simpler if you used `any` https://docs.python.org/2/library/functions.html#any - I don't think we need to know the number of conflicting args or which ones conflict - I think it is sufficient to say "more than one of zone, ipset, or service was specified with present/absent" or something like that

```python
if any((zone, ipset, service)): ... fail ...
```

Reply:
> ```python
> if any((zone, ipset, service)): ... fail ...
> ```

A conditional like this wouldn't work, since it would cause a failure if any of these were set, when the only fail case should be if two or more of these arguments. This is here mostly because I thought it was cleaner than checking for the other two arguments if `zone` is specified, then `ipset` if `service` is set

Reply:
Also, updated the comments to be more clear that those three options are not incompatible with `state: present` and `state: absent` but rather that they are incompatible with one another.